app/vminsert/influx/request_handler.go

```diff
@@ -108,12 +108,14 @@ func insertRows(db string, rows []parser.Row, extraLabels []prompbmarshal.Label)
 				metricGroup := bytesutil.ToUnsafeString(ctx.metricGroupBuf)
 				ic.Labels = append(ic.Labels[:0], ctx.originLabels...)
 				ic.AddLabel("", metricGroup)
-				if err := ic.WriteDataPoint(nil, true, ic.Labels, r.Timestamp, f.Value); err != nil {
+				if !ic.TryPrepareLabels(hasRelabeling) {
+					continue
+				}
+				if err := ic.WriteDataPoint(nil, ic.Labels, r.Timestamp, f.Value); err != nil {
 					return err
 				}
 			}
 		} else {
-			ic.SortLabelsIfNeeded()
```

Comment:
would unsorted labels affect `storage.MarshalMetricNameRaw` on next line in a bad way?

Reply:
good point!